llms.txt

```diff
@@ -11289,6 +11290,780 @@ Doc-Content: https://docs.kluster.ai/tutorials/klusterai-api/image-analysis.ipyn
+    "    \n",
+    "    # Create a new message with the tool results\n",
+    "    messages = [\n",
+    "        {\"role\": \"user\", \"content\": original_query},\n",
+    "        message.model_dump(),\n",
+    "    ]\n",
+    "    \n",
+    "    # Add the tool results\n",
+    "    for result in tool_results:\n",
+    "        messages.append({\n",
+    "            \"role\": \"tool\",\n",
+    "            \"tool_call_id\": result[\"tool_call_id\"],\n",
+    "            \"content\": json.dumps(result[\"result\"])\n",
+    "        })\n",
+    "    \n",
+    "    # Get the final response\n",
+    "    final_response = client.chat.completions.create(\n",
+    "        model=\"klusterai/Meta-Llama-3.1-8B-Instruct-Turbo\",\n",
+    "        messages=messages\n",
+    "    )\n",
+    "    \n",
+    "    return final_response.choices[0].message.content\n",
+    "\n",
+    "# Test with a query that would benefit from web search\n",
+    "search_query = \"What are the latest findings on climate change?\"\n",
+    "search_response = run_with_tools(search_query, web_search_tools)\n",
+    "final_search_result = process_web_search(search_response, search_query)\n",
+    "\n",
+    "print(final_search_result)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Multi-Tool Example\n",
+    "\n",
+    "Real-world questions often need more than one capability. For example, a user might ask “look up Bitcoin’s market cap and convert it to euros.” By registering several tools in the same tools array we give the LLM a menu of options. In the prompt below we include both web_search and calculator, then ask:\n",
+    "\n",
+    "“If Earth’s temperature rises by 2 °C, what percentage increase is that from the current average of 15 °C?”\n",
+    "\n",
+    "The model inspects the schemas and decides this is purely arithmetic, so it emits a single `tool_calls` entry for the calculator. Our `process_multi_tool_calls()` helper loops over each requested tool, dispatches to the matching execution function, bundles every result into `{role:\"tool\"}` messages, and lets the model wrap things up in plain English.\n",
+    "\n",
+    "Had the question also required fresh data, e.g. “…and cite a recent study on global warming”, the LLM could have issued two calls in one turn: first web_search, then calculator. That illustrates the power of multi-tool orchestration: the model can plan a mini-workflow, while your code executes each deterministic step and feeds the results back for seamless narration."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "If the Earth's temperature rises by 2 degrees at a current average global temperature of 15 degrees Celsius, the percentage increase would be approximately 13.33%.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Define multiple tools\n",
```

Comment:
Again I think this giant code block is a bit hard to digest. Is there a way you can recycle functions from before and make it a bit more streamline?